tests/system/test_pandas.py

```diff
@@ -798,3 +798,154 @@ def test_list_rows_max_results_w_bqstorage(bigquery_client):
+        ('baz', null)
+        """
+    ).result()
+    df = bigquery_client.query(
+        f"select * from {dataset_id}.geolake order by name"
+    ).to_geodataframe()
+    assert [v.__class__.__name__ for v in df["geog"]] == [
+        "Polygon",
+        "NoneType",
+        "Point",
+    ]
+    assert df.__class__.__name__ == "GeoDataFrame"
+    assert df["geog"].__class__.__name__ == "GeoSeries"
+    assert list(map(str, df["geog"])) == [
+        "POLYGON ((1 0, 1 1, 0 0, 1 0))",
+        "None",
+        "POINT (0 0)",
+    ]
+    assert str(df.area) == ("0    0.5\n" "1    NaN\n" "2    0.0\n" "dtype: float64")
+    assert df.crs.srs == "EPSG:4326"
+    assert df.crs.name == "WGS 84"
+    assert df.geog.crs.srs == "EPSG:4326"
+    assert df.geog.crs.name == "WGS 84"
+
+
+def test_load_geodataframe(bigquery_client, dataset_id):
+    geopandas = pytest.importorskip("geopandas")
+    import pandas
+    from shapely import wkt
+    from google.cloud.bigquery.schema import SchemaField
+
+    df = geopandas.GeoDataFrame(
+        pandas.DataFrame(
+            dict(
+                name=["foo", "bar"],
+                geo1=[None, None],
+                geo2=[None, wkt.loads("Point(1 1)")],
+            )
+        ),
+        geometry="geo1",
+    )
+
+    table_id = f"{dataset_id}.lake_from_gp"
+    bigquery_client.load_table_from_dataframe(df, table_id).result()
+
+    table = bigquery_client.get_table(table_id)
+    assert table.schema == [
+        SchemaField("name", "STRING", "NULLABLE"),
+        SchemaField("geo1", "GEOGRAPHY", "NULLABLE"),
+        SchemaField("geo2", "GEOGRAPHY", "NULLABLE"),
+    ]
+    assert sorted(map(list, bigquery_client.list_rows(table_id))) == [
+        ["bar", None, "POINT(1 1)"],
+        ["foo", None, None],
+    ]
+
+
+def test_load_dataframe_w_shapely(bigquery_client, dataset_id):
+    pandas = pytest.importorskip("pandas")
+    wkt = pytest.importorskip("shapely.wkt")
+    from google.cloud.bigquery.schema import SchemaField
+
+    df = pandas.DataFrame(
+        dict(name=["foo", "bar"], geo=[None, wkt.loads("Point(1 1)")])
+    )
+
+    table_id = f"{dataset_id}.lake_from_shapes"
+    bigquery_client.load_table_from_dataframe(df, table_id).result()
+
+    table = bigquery_client.get_table(table_id)
+    assert table.schema == [
+        SchemaField("name", "STRING", "NULLABLE"),
+        SchemaField("geo", "GEOGRAPHY", "NULLABLE"),
+    ]
+    assert sorted(map(list, bigquery_client.list_rows(table_id))) == [
+        ["bar", "POINT(1 1)"],
+        ["foo", None],
+    ]
+
+    bigquery_client.load_table_from_dataframe(df, table_id).result()
+    assert sorted(map(list, bigquery_client.list_rows(table_id))) == [
+        ["bar", "POINT(1 1)"],
+        ["bar", "POINT(1 1)"],
+        ["foo", None],
+        ["foo", None],
+    ]
+
+
+def test_load_dataframe_w_wkb(bigquery_client, dataset_id):
+    pandas = pytest.importorskip("pandas")
```

Comment:
We're already requiring pandas for all tests in this module, so I think this line is redundant.

Reply:
Sure. I'll get rid of that.

Reply:
done